Fix Sample Source File Warning Error in Test

This issue suggests fixing the following build error generated after testing the project on macOS using the default compiler:

```cmake
Internal ctest changing into directory: /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/build
Test project /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/build
    Start 1: CheckCoverageTest.cmake
1/1 Test #1: CheckCoverageTest.cmake ..........***Failed    2.10 sec
-- Check test coverage
--   configure sample project
--   build sample project
CMake Error at Assertion.cmake:123 (message):
  expected command:

    /usr/local/Cellar/cmake/3.29.0/bin/cmake --build /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/sample/build

  not to fail with error:

    /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/sample/src/fibonacci.cpp:4:22: warning: generalized initializer lists are a C++11 extension [-Wc++11-extensions]
      if (n <= 0) return {};
                         ^~
    /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/sample/src/fibonacci.cpp:4:22: error: non-aggregate type 'std::vector<int>' cannot be initialized with an initializer list
      if (n <= 0) return {};
                         ^~
    /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/sample/src/fibonacci.cpp:5:22: warning: generalized initializer lists are a C++11 extension [-Wc++11-extensions]
      if (n == 1) return {1};
                         ^~~
    /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/sample/src/fibonacci.cpp:5:22: error: non-aggregate type 'std::vector<int>' cannot be initialized with an initializer list
      if (n == 1) return {1};
                         ^~~
    2 warnings and 2 errors generated.
    make[2]: *** [CMakeFiles/fibonacci.dir/src/fibonacci.cpp.o] Error 1
    make[1]: *** [CMakeFiles/fibonacci.dir/all] Error 2
    make: *** [all] Error 2
    
Call Stack (most recent call first):
  Assertion.cmake:488 (fail)
  /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/CheckCoverageTest.cmake:26 (assert_execute_process)
  /Users/threeal/Workspaces/cmake/CheckCoverage.cmake/test/CheckCoverageTest.cmake:76 (build_sample)




0% tests passed, 1 tests failed out of 1

Total Test time (real) =   2.10 sec

The following tests FAILED:
          1 - CheckCoverageTest.cmake (Failed)
Errors while running CTest
```

The error is likely caused by not setting the C++ standard to C++11 for compiling the source file.

--- a/test/sample/CMakeLists.txt
+++ b/test/sample/CMakeLists.txt
@@ -12,6 +12,7 @@ find_package(
 
 add_library(fibonacci src/fibonacci.cpp)
 target_include_directories(fibonacci PUBLIC include)
+set_property(TARGET fibonacci PROPERTY CXX_STANDARD 11)
 if(NOT WITHOUT_COVERAGE_FLAGS)
   target_check_coverage(fibonacci)
 endif()
@@ -20,6 +21,7 @@ enable_testing()
 
 add_executable(fibonacci_test test/fibonacci_test.cpp)
 target_link_libraries(fibonacci_test PRIVATE fibonacci)
+set_property(TARGET fibonacci_test PROPERTY CXX_STANDARD 11)
 if(NOT WITHOUT_COVERAGE_FLAGS)
   target_check_coverage(fibonacci_test)
 endif()

